tests/test_cases.rs

```diff
@@ -6,18 +6,26 @@ use anyhow::Result;
 use fs_extra::dir::{copy, CopyOptions};
 use std::fs;
 use std::io::{self, Write};
+use log::warn;
 use std::path::Path;
-use std::process::Command;
+use std::process::{Command, exit};
 use tempfile::TempDir;
 
 #[ctor::ctor]
 fn unshorn() {
-    use nix::{mount, sched, unistd};
+    use nix::{errno, mount, sched, unistd};
     use std::os::unix::fs::symlink;
 
     let (uid, gid) = (unistd::geteuid(), unistd::getegid());
     if !uid.is_root() {
-        sched::unshare(sched::CloneFlags::CLONE_NEWUSER).expect("unshare(NEWUSER)");
+        match sched::unshare(sched::CloneFlags::CLONE_NEWUSER) {
+            Ok(_) => {}
+            Err(errno::Errno::EPERM) => {
+                warn!("No permission to unshare namespace, skipping tests");
+                exit(0);
+            }
+            Err(err) => panic!("Unexpected error: {}", err),
```

Comment:
Again, what caused this error? Why not just
```rust
e @ Err(_) => e.expect("unshare(NEWUSER)"),
```
or, indeed, drop the oddly-phrased `Ok(()) => (),` case at the top and replace this line with just
```rust
r => r.expect("unshare(NEWUSER)"),
```